### PR TITLE
Use BinDeps to install freetype

### LINKED ISSUE #17: LoadError: FreeType library could not be found on macOS
Got the following while building FreeType (Julia 0.5, macOS Sierra) :

INFO: Building FreeType
==============================[ ERROR: FreeType ]===============================

LoadError: FreeType library could not be found. Please file an issue here: https://github.com/jhasse/FreeType.jl/issues/new
while loading /Users/igor/.julia/FreeType/deps/build.jl, in expression starting on line 18

Homebrew tells me that freetype-2.6_1 already installed, maybe someone could help?

### PR DESCRIPTION
**Pull request by Gnimuc**
Try to fix #17 and #19 . @SimonDanisch please take a look at this.

cc: @wgreenr @samuelpowell

I have no idea why CI failed on julia-nightly.

### CODE REVIEW

**Comment by samuelpowell** [comment]
Tested successfully on OS X, thus fixes #17

**Comment by SimonDanisch** [comment]
Woops, completely ignored this and just redid the work -.- sorry about that. Let me figure out how to best merge these!

**Comment by SimonDanisch** [comment]
@tkelman, sorry to ping you here, but can you take a quick look if this might cause problem on any more obscure platform? This looks good to me and has passed the test on my available PC's so it seems fine.
This doesn't seem to trigger a `sudo apt-get install` on linux, which is nice, but I'm not sure if that's reliable.
I think freetype should be available per default on most linux system, so people shouldn't need to struggle with: https://github.com/JuliaLang/BinDeps.jl/issues/199! Thanks!!

File: deps/build.jl [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
-        "/usr/lib/i386-linux-gnu", "/usr/lib/x86_64-linux-gnu",
-        Pkg.dir("WinRPM", "deps", "usr", "$(Sys.ARCH)-w64-mingw32", "sys-root", "mingw", "bin"),
-        "/Developer/SDKs/MacOSX10.7.sdk/usr/X11/include/freetype2", "/usr/X11/lib/"
-    ]
-)
-
-if isempty(freetype)
-    error("FreeType library could not be found. Please file an issue here: https://github.com/jhasse/FreeType.jl/issues/new")
+if is_apple()
+    if Pkg.installed("Homebrew") === nothing
```
Comment (tkelman) [inline deps/build.jl]: not necessary

**Comment by tkelman** [comment]
I doubt freetype is present in minimal containers. Should test what this does on systems that don't have providers listed, like Arch or Gentoo which at least a few people run Julia on.

**Comment by SimonDanisch** [comment]
I don't really have access to such a machine :(
When I think about it, it seems to be a fair assumption that someone on such a system is savvy enough to open an issue or figure things out...

**Comment by tkelman** [comment]
Sure, but would be good to avoid regressions. Could use docker or vagrant to test.

**Comment by SimonDanisch** [comment]
Okay, thanks a lot for the confirmation! Well, this shouldn't be a regression, since it was much worse before from what I understand! I'm kinda hung up with other things, and this PR is pretty urgent, so I can't really afford to read up on how to set these things up before merging... :(

**Comment by Gnimuc** [comment]
I'll test this on other platforms(using Docker)  today.

**Comment by Gnimuc** [comment]
Got one problem: 
```julia
# debian
julia> Pkg.build("FreeType")
INFO: Building FreeType
Installing dependency libfreetype6 via `sudo apt-get install libfreetype6`:
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  libpng12-0
The following NEW packages will be installed:
  libfreetype6 libpng12-0
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 639 kB of archives.
After this operation, 1184 kB of additional disk space will be used.
Do you want to continue? [Y/n] Abort.
=================================================================[ ERROR: FreeType ]==================================================================

LoadError: failed process: Process(`sudo apt-get install libfreetype6`, ProcessExited(1)) [1]
while loading /root/.julia/v0.5/FreeType/deps/build.jl, in expression starting on line 25
```
It seems we should find a way to pass `-y` to package manger.

**Comment by tkelman** [comment]
that's probably https://github.com/JuliaLang/BinDeps.jl/issues/199 - it used to be able to switch to letting apt-get prompt, but that no longer works for incomprehensible reasons...

**Comment by Gnimuc** [comment]
@SimonDanisch I've tested the build script on these four platforms: [verbose info here](https://gist.github.com/Gnimuc/6ff93b086f37e05bb2c97af8562f30d3). All of them are suffering from JuliaLang/BinDeps.jl#199, but I think careful users could easily find how to manually install the package from those error strings.

**Comment by SimonDanisch** [comment]
Oh wow, thanks a lot!
Seems like that's as good as it can get at the moment ;)

**Comment by tkelman** [comment]
I'm most interested in what this does on a platform that doesn't have a package manager listed here. Or what users who don't have root access are supposed to do.

**Pull request merged**